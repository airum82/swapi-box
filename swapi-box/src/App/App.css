.App {
  display: grid;
  grid-template-columns: 25% 75%;
  text-align: center;
}


.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}
