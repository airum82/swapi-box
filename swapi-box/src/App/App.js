import React, { Component } from 'react';
import './App.css';
import { Intro } from '../Intro/Intro';
import { FavoritesButton } from '../FavoritesButton/FavoritesButton';
import { PeopleButton } from '../PeopleButton/PeopleButton';
import { PlanetsButton } from '../PlanetsButton/PlanetsButton';
import { VehiclesButton } from '../VehiclesButton/VehiclesButton';
import { CategoryContainer } from '../CategoryContainer/CategoryContainer';
import { Card } from '../Card/Card';
import { helper } from '../helper/helper.js';
import { apiHelper } from '../Api-call/Api-call.js';


class App extends Component {
  constructor() {
    super()
    this.state = {
      helper: new helper(),
      api : new apiHelper(),
      intro: {},
      people: [],
      planets: [],
      vehicles: []

    }
  }

  resetState = () => {
    this.setState({
      people: [],
      planets: [],
      vehicles: []
    })
  }

  retrieveData = (data, sampleItem) => {
    this.resetState()
    let category;
    if(Object.keys(sampleItem).includes('residents')) {
      category = 'planets';
    } else if(Object.keys(sampleItem).includes('model')) {
      category = 'vehicles';
    } else {
      category = 'people'
    }
    this.setState({
      [category]: data
    })
  }

  componentDidMount() {
    // const url = "https://swapi.co/api/films/";
    // fetch(url)
    // .then(response => response.json())
    // .then(result => this.state.helper.pickFilmIntro(result))
    // .then(intro => this.setState({ intro }))
  }

  render() {
    let category;
    if(this.state.planets.length) {
      category = this.state.planets;
    } else if(this.state.people.length) {
      category = this.state.people;
    } else {
      category = this.state.vehicles
    }
    return (
      <div className="App">
        <Intro intro={this.state.intro}/>
        <main className='main'>
          <h1>Swapi-box</h1>
          <hr />
          <section className="buttons">
            <PeopleButton 
              viewPeople={this.state.api.viewPeople}
              cleanPeople={this.state.helper.cleanPeople}
              retrieveData={this.retrieveData}
            />
            <PlanetsButton 
              fetchPlanets={this.state.api.fetchPlanets}
              retrieveData={this.retrieveData}
              cleanPlanetData={this.state.helper.cleanPlanetData}
            />
            <VehiclesButton 
              fetchVehicles={this.state.api.fetchVehicles}
              cleanVehicles={this.state.helper.cleanVehicles}
              retrieveVehicles={this.retrieveData} 
            />
            <FavoritesButton/>
          </section>
          <CategoryContainer category={category}/>
        </main>
      </div>
    );
  }
}

export default App;
