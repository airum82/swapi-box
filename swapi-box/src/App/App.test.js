import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot upon initial render', () => {
    const wrapper = shallow(<App/>, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  })

  it('reset state should reset the state to state upon render', () => {
    const wrapper = shallow(<App/>, { disableLifecycleMethods: true });
    const initialState = wrapper.state();
    wrapper.instance().setState({ planets: ['Venus', 'Mercury', 'Saturn'],
                                  vehicles: ['Audi', 'Prius'] });
    wrapper.instance().resetState();

    expect(wrapper.state()).toEqual(initialState);
  })

  it('retrieve data should call resetState', () => {
    const wrapper = shallow(<App/>, { disableLifecycleMethods: true });
    const mockData = ['this', 'that', 'some']
    wrapper.instance().resetState = jest.fn();
    wrapper.instance().retrieveData(mockData, mockData[0]);
    expect(wrapper.instance().resetState).toHaveBeenCalled();
  })

  it('retrieveData should call determineCategory with correct params', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true});
    const mockData = ['this', 'that', 'some']
    wrapper.instance().determineCategory = jest.fn();
    wrapper.instance().retrieveData(mockData, mockData[0]);
    expect(wrapper.instance().determineCategory).toHaveBeenCalledWith(mockData[0]);
  })

  it('favoriteCard should call determine category with correct params', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true});
    wrapper.instance().determineCategory = jest.fn();
    wrapper.instance().setState({ planets: ['whatever', 'wherever'] });
    const mockProps = {name: 'asdas',
                       residents: 'rocky'};
    wrapper.instance().favoriteCard(mockProps);
    expect(wrapper.instance().determineCategory).toHaveBeenCalledWith(mockProps)

  })
  
})

