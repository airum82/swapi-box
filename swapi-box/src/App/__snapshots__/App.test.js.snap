// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot upon initial render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          intro={Object {}}
/>,
        <main
          className="main"
>
          <h1>
                    Swapi-box
          </h1>
          <hr />
          <section
                    className="buttons"
          >
                    <Unknown
                              cleanPeople={[Function]}
                              retrieveData={[Function]}
                              viewPeople={[Function]}
                    />
                    <Unknown
                              cleanPlanetData={[Function]}
                              fetchPlanets={[Function]}
                              retrieveData={[Function]}
                    />
                    <Unknown
                              cleanVehicles={[Function]}
                              fetchVehicles={[Function]}
                              retrieveVehicles={[Function]}
                    />
                    <FavoritesButton />
          </section>
          <Unknown
                    category={Array []}
                    favoriteCard={[Function]}
          />
</main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "intro": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Swapi-box
</h1>,
            <hr />,
            <section
              className="buttons"
>
              <Unknown
                            cleanPeople={[Function]}
                            retrieveData={[Function]}
                            viewPeople={[Function]}
              />
              <Unknown
                            cleanPlanetData={[Function]}
                            fetchPlanets={[Function]}
                            retrieveData={[Function]}
              />
              <Unknown
                            cleanVehicles={[Function]}
                            fetchVehicles={[Function]}
                            retrieveVehicles={[Function]}
              />
              <FavoritesButton />
</section>,
            <Unknown
              category={Array []}
              favoriteCard={[Function]}
/>,
          ],
          "className": "main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Swapi-box",
            },
            "ref": null,
            "rendered": "Swapi-box",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Unknown
                  cleanPeople={[Function]}
                  retrieveData={[Function]}
                  viewPeople={[Function]}
/>,
                <Unknown
                  cleanPlanetData={[Function]}
                  fetchPlanets={[Function]}
                  retrieveData={[Function]}
/>,
                <Unknown
                  cleanVehicles={[Function]}
                  fetchVehicles={[Function]}
                  retrieveVehicles={[Function]}
/>,
                <FavoritesButton />,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "cleanPeople": [Function],
                  "retrieveData": [Function],
                  "viewPeople": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "cleanPlanetData": [Function],
                  "fetchPlanets": [Function],
                  "retrieveData": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "cleanVehicles": [Function],
                  "fetchVehicles": [Function],
                  "retrieveVehicles": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": Array [],
              "favoriteCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            intro={Object {}}
/>,
          <main
            className="main"
>
            <h1>
                        Swapi-box
            </h1>
            <hr />
            <section
                        className="buttons"
            >
                        <Unknown
                                    cleanPeople={[Function]}
                                    retrieveData={[Function]}
                                    viewPeople={[Function]}
                        />
                        <Unknown
                                    cleanPlanetData={[Function]}
                                    fetchPlanets={[Function]}
                                    retrieveData={[Function]}
                        />
                        <Unknown
                                    cleanVehicles={[Function]}
                                    fetchVehicles={[Function]}
                                    retrieveVehicles={[Function]}
                        />
                        <FavoritesButton />
            </section>
            <Unknown
                        category={Array []}
                        favoriteCard={[Function]}
            />
</main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "intro": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Swapi-box
</h1>,
              <hr />,
              <section
                className="buttons"
>
                <Unknown
                                cleanPeople={[Function]}
                                retrieveData={[Function]}
                                viewPeople={[Function]}
                />
                <Unknown
                                cleanPlanetData={[Function]}
                                fetchPlanets={[Function]}
                                retrieveData={[Function]}
                />
                <Unknown
                                cleanVehicles={[Function]}
                                fetchVehicles={[Function]}
                                retrieveVehicles={[Function]}
                />
                <FavoritesButton />
</section>,
              <Unknown
                category={Array []}
                favoriteCard={[Function]}
/>,
            ],
            "className": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Swapi-box",
              },
              "ref": null,
              "rendered": "Swapi-box",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Unknown
                    cleanPeople={[Function]}
                    retrieveData={[Function]}
                    viewPeople={[Function]}
/>,
                  <Unknown
                    cleanPlanetData={[Function]}
                    fetchPlanets={[Function]}
                    retrieveData={[Function]}
/>,
                  <Unknown
                    cleanVehicles={[Function]}
                    fetchVehicles={[Function]}
                    retrieveVehicles={[Function]}
/>,
                  <FavoritesButton />,
                ],
                "className": "buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "cleanPeople": [Function],
                    "retrieveData": [Function],
                    "viewPeople": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "cleanPlanetData": [Function],
                    "fetchPlanets": [Function],
                    "retrieveData": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "cleanVehicles": [Function],
                    "fetchVehicles": [Function],
                    "retrieveVehicles": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "category": Array [],
                "favoriteCard": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
